src/overlays/effects/ovl_Effect_Ss_Dead_Ds/z_eff_ss_dead_ds.c

```diff
@@ -1,27 +1,124 @@
 void EffectSsDeadDs_Draw(PlayState* play, u32 index, EffectSs* this);
 
-#if 0
 const EffectSsInit Effect_Ss_Dead_Ds_InitVars = {
     EFFECT_SS_DEAD_DS,
     EffectSsDeadDs_Init,
 };
 
-#endif
+u32 EffectSsDeadDs_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx) {
+    EffectSsDeadDsInitParams* initParams = PARAMS;
+
+    this->pos = initParams->pos;
+    this->velocity = initParams->velocity;
+    this->accel = initParams->accel;
+    this->life = initParams->life;
+    this->rScaleStep = initParams->scaleStep;
+    this->rHalfOfLife = initParams->life / 2;
+    this->rAlphaStep = initParams->alpha / this->rHalfOfLife;
+    this->draw = EffectSsDeadDs_Draw;
+    this->update = EffectSsDeadDs_Update;
+    this->rScale = initParams->scale;
+    this->rAlpha = initParams->alpha;
+    this->rTimer = 0;
+
+    return 1;
+}
+
+void EffectSsDeadDs_Draw(PlayState* play, u32 index, EffectSs* this) {
+    s32 pad;
+    f32 scale;
+    s32 pad2[2];
+    MtxF mf;
+    f32 yIntersect;
+    Vec3f pos;
+    CollisionPoly* floorPoly;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    scale = this->rScale * 0.01f;
+    func_8012C974(play->state.gfxCtx);
+    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 0, 0, 0, this->rAlpha);
+    gDPSetEnvColor(POLY_XLU_DISP++, 0, 0, 0, 0);
+    pos = this->pos;
+
+    if (this->rTimer == 0) {
+        Vec3s rot;
+        Vec3f prevPos;
+
+        prevPos.x = pos.x - this->velocity.x;
+        prevPos.y = pos.y - this->velocity.y;
+        prevPos.z = pos.z - this->velocity.z;
+
+        if (BgCheck_EntitySphVsWall1(&play->colCtx, &this->pos, &pos, &prevPos, 1.5f, &floorPoly, 1.0f)) {
+            func_800C0094(floorPoly, this->pos.x, this->pos.y, this->pos.z, &mf);
+            Matrix_Put(&mf);
+        } else {
+            pos.y++;
+            yIntersect = BgCheck_EntityRaycastFloor1(&play->colCtx, &floorPoly, &pos);
+
+            if (floorPoly != NULL) {
+                func_800C0094(floorPoly, this->pos.x, yIntersect + 1.5f, this->pos.z, &mf);
+                Matrix_Put(&mf);
+            } else {
+                Matrix_Translate(this->pos.x, this->pos.y, this->pos.z, MTXMODE_NEW);
+                Matrix_Get(&mf);
+            }
+        }
+
+        Matrix_MtxFToZYXRot(&mf, &rot, false);
+        this->rRoll = rot.x;
+        this->rPitch = rot.y;
+        this->rYaw = rot.z;
+        this->pos.y = mf.yw;
+        this->rTimer++;
+    }
+
+    Matrix_Translate(this->pos.x, this->pos.y, this->pos.z, MTXMODE_NEW);
+    Matrix_RotateZYX(this->rRoll, this->rPitch, this->rYaw, MTXMODE_APPLY);
+    Matrix_RotateXFApply(1.57f);
+    Matrix_Scale(scale, scale, scale, MTXMODE_APPLY);
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gDPSetCombineLERP(POLY_XLU_DISP++, 0, 0, 0, PRIMITIVE, TEXEL0, 0, PRIMITIVE, 0, 0, 0, 0, PRIMITIVE, TEXEL0, 0,
+                      PRIMITIVE, 0);
+    gSPDisplayList(POLY_XLU_DISP++, gLensFlareCircleDL);
```

Comment:
Hmm, this name may need a rethink, although I'm guessing it's used elsewhere.

Reply:
Based on OoT used in Kankyo.

Reply:
Seems like they really liked reusing the lens textures/display lists for other things in MM. I know we currently switched `gLensOfTruthMaskTex` -> `gCircleTex` (though that name is probably far to generic tbh, might also need a rethink), but yeah, not sure what to do with these. Might not be worth rethinking until we have all use-cases (near 100% decomp) so we can get a bigger picture.

Reply:
Pretty sure that is something different. This is part of gameplay keep and looks to be used in just `DeadDd`, `DeadDs`, and `kankyo`.